lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -764,7 +770,13 @@ def pass_sanity_hook_checks(self, event: EventBase) -> bool:
             return False
 
         if not self.charm.is_relation_feasible(self.relation_name):
-            logger.info("Skipping event %s , relation not feasible.", str(type(event)))
+            if self.charm.status.is_status_related_to_mismatched_revision(
+                self.charm.unit.status.name
+            ):
+                logger.info("Deferring %s. Mismatched versions in the cluster.", str(type(event)))
+                event.defer()
+            else:
+                logger.info("Skipping event %s , relation not feasible.", str(type(event)))
```

Comment:
Nit: Removing nested structure of if statement
```suggestion
      if self.charm.status.is_status_related_to_mismatched_revision(
           self.charm.unit.status.name
      ):
            logger.info("Deferring %s. Mismatched versions in the cluster.", str(type(event)))
            event.defer()
       elif not self.charm.is_relation_feasible(self.relation_name):
           logger.info("Skipping event %s , relation not feasible.", str(type(event)))
```

Reply:
(Same comment as the former one)